src/main/java/com/pirimid/cryptotrade/websocket/coinbase/handler/CoinbaseSessionHandler.java

```diff
@@ -0,0 +1,160 @@
+import com.pirimid.cryptotrade.websocket.coinbase.res.WSCoinbaseOrderDto;
+import com.pirimid.cryptotrade.websocket.coinbase.res.WSCoinbaseTradeDto;
+import org.springframework.web.socket.CloseStatus;
+import org.springframework.web.socket.TextMessage;
+import org.springframework.web.socket.WebSocketHandler;
+import org.springframework.web.socket.WebSocketMessage;
+import org.springframework.web.socket.WebSocketSession;
+
+import java.io.IOException;
+import java.security.InvalidKeyException;
+import java.security.NoSuchAlgorithmException;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+
+public class CoinbaseSessionHandler implements WebSocketHandler {
+
+    private Account account;
+    private OrderService orderService;
+    private Boolean isConnected = false;
+    private WebSocketSession session;
+    private WSCoinbase coinbase;
+    Gson gson = new Gson();
+
+
+    public CoinbaseSessionHandler(Account account, OrderService orderService, WSCoinbase coinbase) {
+        this.account = account;
+        this.orderService = orderService;
+        this.coinbase=coinbase;
+    }
+
+    private void sendData(String data) throws IOException {
+        WebSocketMessage webSocketMessage = new TextMessage(data);
+        session.sendMessage(webSocketMessage);
+    }
+
+    private void fullChannelWithAuth(String key, String secretKey, String passphrase) throws IOException, NoSuchAlgorithmException, InvalidKeyException {
+        Long dateInL = Instant.now().getEpochSecond();
+        Date date = new Date(dateInL);
+        String signature = CoinbaseUtil.getSignature(String.valueOf(dateInL), secretKey, "GET", "/users/self/verify", "");
+        List<ReqChannel> channels = new ArrayList<>();
+        channels.add(ReqChannel.USER);
+        channels.add(ReqChannel.HEARTBEAT);
+        List<String> pids = new ArrayList<>();
+        pids.add("BTC-USD");
+        ChannelReq req = ChannelReq.builder()
+                .type(ReqType.SUBSCRIBE)
+                .productIds(pids)
+                .channels(channels)
+                .key(key)
+                .passphrase(passphrase)
+                .signature(signature)
+                .timestamp(date)
+                .build();
+        ChannelReq req1 = new ChannelReq();
+
+        ObjectMapper objectMapper = new ObjectMapper();
+        String data = objectMapper.writeValueAsString(req);
+        sendData(data);
+
+    }
+
+    @Override
+    public void afterConnectionEstablished(WebSocketSession session) throws Exception {
+        this.session = session;
+        try {
+            fullChannelWithAuth(account.getApiKey(), account.getSecretKey(), account.getPassPhrase());
+        } catch (IOException e) {
+            e.printStackTrace();
+        } catch (NoSuchAlgorithmException e) {
+            e.printStackTrace();
+        } catch (InvalidKeyException e) {
+            e.printStackTrace();
+        }
```

Comment:
Can we add generic catch?

Reply:
We are planning to add a centralized exception class once we are done with some front-end. This class will be responsible for sending appropriate response to the front-end based on the type of exception